sound/soc/codecs/rt700.c

```diff
@@ -307,6 +310,10 @@ static void rt700_btn_check_handler(struct work_struct *work)
 			SND_JACK_HEADSET |
 			SND_JACK_BTN_0 | SND_JACK_BTN_1 |
 			SND_JACK_BTN_2 | SND_JACK_BTN_3);
+	snd_soc_jack_report(rt700->hs_jack, rt700->jack_type,
+			SND_JACK_HEADSET |
+			SND_JACK_BTN_0 | SND_JACK_BTN_1 |
+			SND_JACK_BTN_2 | SND_JACK_BTN_3);
```

Comment:
same here, what does this do?

Reply:
I just want the button response like click one by one.
If a user keeps pressing the button, the system will report for volume+/- too quickly.
If this modification is not well, another way I could adjust the delay time of the work queue.

Reply:
I see. The issue I have is that not all buttons are mapped to volume...
How is this solved with other codecs, it doesn't look like an rt700 issue only, does it?

Reply:
Yeah, I checked the source code how to do in HDA mode.
The snd_hda_jack_report_sync function in sound/pci/hda/hda_jack.c reports twice if it has the button_state.

Reply:
did you mean the code below
````c
state = jack->button_state;
			if (get_jack_plug_state(jack->pin_sense))
				state |= jack->type;
			snd_jack_report(jack->jack, state);
			if (jack->button_state) {
				snd_jack_report(jack->jack,
						state & ~jack->button_state);
				jack->button_state = 0; /* button released */
			}
````

If yes, then you should add a comment that this deals with button released or explain why you have two calls which seem to do the same thing.

Reply:
Yes, I will make a patch for button issue and resend the patches.